OpenRA.Mods.Common/ServerTraits/MasterServerPinger.cs

```diff
@@ -82,12 +82,22 @@ public void Tick(S server)
 					server.SendMessage(masterServerMessages.Dequeue());
 		}
 
-		public void ServerStarted(S server)
+		void INotifyServerStart.ServerStarted(S server)
 		{
 			if (server.Type != ServerType.Local && LanGameBeacon != null)
 				LanGameBeacon.Start();
 		}
 
+		void INotifyServerShutdown.ServerShutdown(S server)
+		{
+			var gameServer = new GameServer(server);
```

Comment:
This can then go inside the `if` and please add a comment explaining that this is notifying the master server to remove the game from the list.